platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStore.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service;
+
+import com.hedera.hapi.node.state.roster.Roster;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Provides read-only methods for interacting with Rosters.
+ */
+public interface ReadableRosterStore {
+
+    /**
+     * Set the candidate roster.
+     *
+     * @param candidateRoster a candidate roster to set
+     */
+    void setCandidateRoster(@NonNull final Roster candidateRoster);
```

Comment:
Why are there setters in a read-only interface?

Reply:
the `ReadableRosterStore` is the base impl which `WritableRosterStore` extends so I have to implement this `setCandidateRoster` which is why this implementation throws an exception. The actual implementation is in the top level `WritableRosterStore`.

Reply:
This is bad practice. Read only interfaces should come with read only implementations. The modification methods should be defined in the WriableRosterStore interface/class.